refined/shared/src/main/scala/monocle/refined/strings.scala

```diff
@@ -1,17 +1,24 @@
 package monocle.refined
 
-import eu.timepit.refined.api.Refined
-import eu.timepit.refined.string.StartsWith
+import eu.timepit.refined.api.{Refined, Validate}
+import eu.timepit.refined.string.{EndsWith, StartsWith}
 import monocle._
 
 object strings extends StringsInstances
 
 trait StringsInstances {
 
-  def startsWith(prefix: String): Prism[String, StartsWithString[prefix.type]] = {
-    Prism.partial[String, Refined[String, StartsWith[prefix.type]]] {
-      case string if string.startsWith(prefix) => Refined.unsafeApply(string)
-    }{_.value}
-  }
+  def startsWith(prefix: String)(implicit v: Validate[String, StartsWith[prefix.type]]): Prism[String, StartsWithString[prefix.type]] =
+    refinedPrism[String, StartsWith[prefix.type]](prefix)
+
+  def endsWith(suffix: String)(implicit v: Validate[String, EndsWith[suffix.type]]): Prism[String, EndsWithString[suffix.type]] =
+    refinedPrism[String, EndsWith[suffix.type]](suffix)
 
+  private def refinedPrism[T, P](t: T)(implicit v: Validate[T, P]): Prism[T, T Refined P] = {
```

Comment:
Btw, if we make this public, people can use it for their own custom predicates that are not in refined.